### PR TITLE
feat(ui-react): implement role-based permission system

### PR DESCRIPTION
**Pull request by luizhf42**
## What

Introduces a centralised RBAC system for the React console that gates UI actions based on the current user's namespace role. Replaces the scattered ad-hoc role derivation (member-lookup chains, `isOwner` flags) that existed in Settings and BannerEdit with a single source of truth.

## Why

The Vue UI already enforces permissions visually. The React console had none, so any authenticated user could see and attempt all action buttons regardless of their role — relying entirely on server 403s for feedback, which is a poor UX.

## Changes

- **`utils/permission.ts`**: `hasPermission(role, action)` pure function with a four-level hierarchy (observer < operator < administrator < owner). `isValidRole` type-guard for safe narrowing of API-sourced strings.

- **`hooks/useHasPermission.ts`**: thin hook wiring the pure utility to the Zustand auth store.

- **`components/common/RestrictedAction.tsx`**: wrapper that, when the user lacks permission, renders children as visually dimmed and fully inert (`pointer-events-none` + HTML `inert` attribute to block keyboard activation, `aria-disabled`, native `title` tooltip). When permitted, renders children unwrapped.

- **13 pages/components**: action buttons (destructive, write operations) wrapped in `RestrictedAction`; inline edit affordances (pencil icons, tag × buttons) conditionally hidden — the two are intentionally different UX patterns.

- **`authStore.ts`**: `role` field narrowed from `string | null` to `Role | null`.

- **`namespacesStore.ts`**: `fetch()` now calls `getNamespaceToken` on every cold start so `authStore.role` is populated from the server before any page renders. Without this, `role` was `null` on first login and every `RestrictedAction` appeared locked — including the owner's "Delete Namespace" button, which fell back to "Leave Namespace". Also keeps the role fresh after an out-of-band role change (previously the persisted localStorage value would be stale).

## Testing

First-login regression: clear localStorage, log in as an owner, navigate to Settings — the Danger Zone should show "Delete Namespace", not "Leave Namespace".

Role boundary: log in as an operator and confirm that Add Device, Accept, and Reject are enabled but Delete Device, Create Public Key, and Delete Namespace are visually disabled with the tooltip. Keyboard-tab to a disabled button and press Enter — it must not fire.

### CODE REVIEW

File: ui-react/apps/console/src/utils/__tests__/permission.test.ts [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+    });
+  });
+
+  describe("boundary conditions — minimum required role", () => {
+    it.each<[Action, Role, Role]>([
+      ["device:accept",    "observer",      "operator"],
+      ["device:remove",    "operator",      "administrator"],
+      ["namespace:delete", "administrator", "owner"],
+      ["billing:subscribe","administrator", "owner"],
+    ])("%s: %s denied, %s allowed", (action, denied, allowed) => {
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/console/src/utils/__tests__/permission.test.ts:183]: Fixed. Now CI is passing.

File: ui-react/apps/console/src/utils/permission.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+
+const roleLevels: Record<Role, RoleLevel> = {
+  observer: RoleLevel.OBSERVER,
+  operator: RoleLevel.OPERATOR,
+  administrator: RoleLevel.ADMINISTRATOR,
+  owner: RoleLevel.OWNER,
+};
+
+export const isValidRole = (role: string | null): role is Role =>
+  role !== null && role in roleLevels;
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/console/src/utils/permission.ts]: Addressed in 234ebdd36.

File: ui-react/apps/console/src/stores/namespacesStore.ts [hunk 3]
```diff
@@ -45,6 +46,22 @@ export const useNamespacesStore = create<NamespacesState>((set) => ({
...
+      // Sync the role for the current tenant on every app startup so it is
+      // always fresh (covers first login and stale-localStorage scenarios).
+      const { tenant } = useAuthStore.getState();
+      if (tenant) {
+        try {
+          const { token, role } = await getNamespaceToken(tenant);
+          useAuthStore.getState().setSession({
+            token,
+            tenant,
+            role: isValidRole(role) ? role : undefined,
```
[the remark this answers is not in the document]
Comment (luizhf42) [reply ui-react/apps/console/src/stores/namespacesStore.ts]: No longer applicable — `namespacesStore.ts` was deleted as part of the TanStack Query migration. The role is now initialized in `useInitRole` (`hooks/useNamespaces.ts`) using the SDK-typed `data.role: NamespaceMemberRole` directly, so no `isValidRole` guard or `undefined` fallback is involved.
